### PR TITLE
Fix permissions to run e2e tests on OCP with arbitrary user ID

### LINKED ISSUE #6977: Not enough permissions to run e2e-tests on OCP with random user ID
Since #6959, no e2e-tests are run on OCP.

It's because OCP runs containers with an arbitrarily assigned user ID.

To reproduce:
```sh
> docker run --rm -ti -e E2E_JSON=true -e E2E_TAGS=e2e,release -u 1000710000 docker.elastic.co/eck-ci/eck-e2e-tests:e2d11c4b
open /.cache/go-build/50/50a175ed06189c9f1c94d7f76047fab9a343f876902ffc803b70cdf4606af522-d: permission denied
...
open /.cache/go-build/a4/a41d67c6132a1bcbb057dfca89697c6bf84cc60cff6b13afbff6d7d0f9d63b20-d: permission denied
```

Two problems:
- `go test` wants to write to `/.cache/go-build/*/*`.
- `run.sh` wants to write to `/go/src/github.com/elastic/cloud-on-k8s` the e2e-tests json results file.

### PR DESCRIPTION
**Pull request by thbkrkr**
This adds write permissions to `/.cache/go-build/*` and `/go/src/github.com/elastic/cloud-on-k8s` after `go generate` was executed in the Dockerfile for the e2e-tests image in order to make it possible to run `test/e2e/run.sh` with an arbitrary user ID.

Resolves #6977.

### CODE REVIEW

**Comment by thbkrkr** [comment]
buildkite test this -f p=ocp,t=TestSmoke

**Comment by thbkrkr** [comment]
buildkite test this -f p=ocp,t=TestSmoke

**Review by naemono — APPROVED** [review]
I typically would cringe at 777 but since these are the e2e tests i feel it's fine

File: test/e2e/Dockerfile [hunk 1]
```diff
@@ -31,4 +31,6 @@ USER 1001
 COPY test/e2e/license.key /license.key
 RUN LICENSE_PUBKEY=/license.key go generate -tags="${GO_TAGS}" ./pkg/... ./cmd/...
 
+RUN chmod 777 /.cache/go-build/* /go/src/github.com/elastic/cloud-on-k8s
```
[the remark this answers is not in the document]
Comment (pebrc) [reply test/e2e/Dockerfile:34]: I like this suggestion

**Review by pebrc — APPROVED** [review]

File: test/e2e/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (thbkrkr) [reply test/e2e/Dockerfile]: Me too. I made the change.

**Comment by thbkrkr** [comment]
buildkite test this -f p=ocp,t=TestSmoke

**Pull request merged**